### PR TITLE
unit separator

### PR DESCRIPTION
**Pull request by Araq**
With this PR the Nim compiler uses the ASCII unit separator before a newline for every end of any kind of message. This is simpler than JSON output, mostly backwards compatible and allows tools to know precisely when one error message starts and the other ends.

### CODE REVIEW

File: testament/important_packages.nim [hunk 1]
```diff
@@ -74,7 +74,7 @@ pkg "gnuplot", "nim c gnuplot.nim"
 pkg "hts", "nim c -o:htss src/hts.nim"
 pkg "httpauth"
 pkg "illwill", "nimble examples"
-pkg "inim"
+#pkg "inim"
```
Comment (Clyybber) [inline testament/important_packages.nim]:
```suggestion
pkg "inim", allowFailure = true
```

File: compiler/msgs.nim [hunk 2]
```diff
@@ -545,14 +554,14 @@ proc liMessage*(conf: ConfigRef; info: TLineInfo, msg: TMsgKind, arg: string,
       if msg == hintProcessing:
         msgWrite(conf, ".")
       else:
-        styledMsgWriteln(styleBright, loc, resetStyle, color, title, resetStyle, s, KindColor, kindmsg)
+        styledMsgWriteln(styleBright, loc, resetStyle, color, title, resetStyle, s, KindColor, kindmsg, UnitSep)
         if conf.hasHint(hintSource) and info != unknownLineInfo:
           conf.writeSurroundingSrc(info)
         if hintMsgOrigin in conf.mainPackageNotes:
```
Comment (timotheecour) [inline compiler/msgs.nim:560]: with `--hint:source` it seems to split at the wrong place, please check

File: testament/important_packages.nim (hunk 1 above)
Comment (Araq) [reply testament/important_packages.nim]: What's the point of `allowFailure`? It's a mess --- unreadable logs, you don't know which package *really* failed, you don't know when it starts to work again and then when it starts to fail again. The current state of failing projects should be reflected in important_packages.nim, not hidden in some enormous CI log.

Comment (Clyybber) [reply testament/important_packages.nim]: With #17757 they would be all in one batch so wouldn't spam the other logs; and eventually we can have the CI notify us when a package starts to work again.

Comment (Araq) [reply testament/important_packages.nim]: Did it this way anyway.

File: compiler/msgs.nim (hunk 2 above)
Comment (Araq) [reply compiler/msgs.nim:560]: Good catch, fixed.

**Pull request merged**